attribute.go

```diff
@@ -296,3 +296,132 @@ func (ad *AttributeDecoder) Do(fn func(b []byte) error) {
 		ad.err = err
 	}
 }
+
+// An AttributeEncoder provides a safe way to encode attributes.
+//
+// It is recommended to use an AttributeEncoder where possible instead of
+// calling MarshalBinary and using package nlenc directly.
+//
+// Errors from intermediate encoding steps are returned in the call to
+// MarshalBinary.
+type AttributeEncoder struct {
+	// ByteOrder defines a specific byte order to use when processing integer
+	// attributes.  ByteOrder should be set immediately after creating the
+	// AttributeEncoder: before any attributes are encoded.
+	//
+	// If not set, the native byte order will be used.
+	ByteOrder binary.ByteOrder
+
+	attrs []Attribute
+	err   error
+}
+
+// NewAttributeEncoder creates an AttributeEncoder that encodes Attributes.
+func NewAttributeEncoder() (*AttributeEncoder, error) {
```

Comment:
Can probably remove the error return here.  Then you could arguably remove the entire constructor.

There is the Go proverb of "make the zero value useful" which could apply well here:

```go
var ae netlink.AttributeEncoder
```

But then we also have to lazily set byte order on first call if that's the case. What do you think?

Reply:
I would normally agree with you. However, here I prefer the symmetry with AttributeDecoder.

Reply:
That's fair.  I was thinking about doing something clever like memoizing byte order on the first call but it makes the internal code less clean.  How about we just remove the error return?